iroh-net/src/hp/netcheck.rs

```diff
@@ -954,38 +618,34 @@ impl Actor {
         }
         inc!(NetcheckMetrics::Reports);
 
-        // TODO: not sure we're allowed to await in this function...  We can make the
-        // probeplan inside the ReportState actor though.
-        let if_state = interfaces::State::new().await;
-        let last = self.reports.last.clone();
-        let plan = ProbePlan::new(&derp_map, &if_state, last.as_deref());
-
-        let actor = reportstate::ReportState::new(
+        let actor = reportgen::Client::new(
             self.addr(),
-            last.clone(),
-            plan,
+            self.reports.last.clone(),
             self.port_mapper.clone(),
             self.skip_external_network,
-            last.is_some(), // TODO: doesn't need to be passed
             derp_map,
             stun_sock_v4,
             stun_sock_v6,
         );
 
-        self.current_report_run = Some((actor, cancel_token.drop_guard(), response_tx));
+        self.current_report_run = Some(ReportRun {
+            _reportgen: actor,
```

Comment:
why `_` prefix?

Reply:
It's described in the doc comment of `ReportRun`.  We're only keeping this as a drop_guard: if dropped the reportgen actor will stop.  So we never access this field.